lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -413,6 +413,14 @@ bool SPIRVRegularizeLLVMBase::regularize() {
           if (isa<PossiblyExactOperator>(BO) && BO->isExact())
             BO->setIsExact(false);
         }
+
+        // FIXME: This is not valid handling for freeze instruction
+        if (auto *FI = dyn_cast<FreezeInst>(&II)) {
```

Comment:
Please also check for undef/poison values and materialize them

Reply:
Updated PR to materialize frozen undef/poison with a random constant.